### PR TITLE
Add support for `activerecord` 5.2.2

### PR DESCRIPTION
**Pull request by tenshiAMD**
In relation to the changes in `activerecord` version `5.2.2`in using `scope_for_create` (https://github.com/rails/rails/pull/33639/files), we must use the new method `values_for_create` provided.

This will resolves the issue https://github.com/krautcomputing/active_record-acts_as/issues/27.

### CODE REVIEW

File: lib/active_record/acts_as/querying.rb [hunk 1]
```diff
@@ -28,10 +28,9 @@ def where!(opts, *rest)
 
     module ScopeForCreate
       def scope_for_create(attributes = nil)
-        scope = ActiveRecord.version.to_s.to_f >= 5.2 ? super(attributes) : where_values_hash
-        if acting_as?
-          scope.merge!(where_values_hash(acting_as_model.table_name))
-        end
+        scope = where_values_hash
+        scope = values_for_create(attributes) if respond_to?(:values_for_create)
```
Comment (manuelmeurer) [inline lib/active_record/acts_as/querying.rb]: Could you combine these lines to `scope = respond_to?(:values_for_create) ? values_for_create(attributes) : where_values_hash`?

File: .travis.yml [hunk 2]
```diff
@@ -5,8 +5,8 @@ rvm:
   - 2.3.4
   - 2.4.1
 before_install:
-  - gem update --system
-  - gem update bundler
+  - gem update --system 2.7.8
+  - gem install bundler -v '~> 1.17.3'
```
Comment (manuelmeurer) [inline .travis.yml]: Why are these changes necessary?

**Review by manuelmeurer — CHANGES_REQUESTED** [review]
Could you also update the Appraisals file to Rails 5.2.2 to check if the tests actually fail without your changes?

**Comment by manuelmeurer** [comment]
Thanks for working on this, @tenshiAMD! 😃

File: lib/active_record/acts_as/querying.rb (hunk 1 above)
Comment (tenshiAMD) [reply lib/active_record/acts_as/querying.rb]: Are your sure, @manuelmeurer? According to `rubocop`, the code below is against its guidelines:

```
scope = respond_to?(:values_for_create) ? values_for_create(attributes) : where_values_hash
```

File: .travis.yml (hunk 2 above)
Comment (tenshiAMD) [reply .travis.yml]: I noticed the ruby versions in `travisCI` config are failing in the latest `rubygems` when running ``gem update --system`` and the bundler it also fails in the latest `2.0` update when ``gem update bundler`` in the previous configuration. So I need to adjust the configuration to pass the tests, you can see the test history below that fails in relation to this:
- https://travis-ci.org/krautcomputing/active_record-acts_as/builds/474851341?utm_source=github_status&utm_medium=notification
- https://travis-ci.org/krautcomputing/active_record-acts_as/builds/474856183?utm_source=github_status&utm_medium=notification

File: lib/active_record/acts_as/querying.rb (hunk 1 above)
Comment (manuelmeurer) [reply lib/active_record/acts_as/querying.rb]: The let's use a standard `if else`. :)
Assigning `scope` and then reassigning it on the next line is not good style IMO.

File: .travis.yml (hunk 2 above)
Comment (manuelmeurer) [reply .travis.yml]: Then let's drop Ruby 2.2 support instead, it's not supported anymore anyways.
I'll make the commit, could you remove these changes to `.travis.yml` and rebase against master?

File: lib/active_record/acts_as/querying.rb (hunk 1 above)
Comment (tenshiAMD) [reply lib/active_record/acts_as/querying.rb]: @manuelmeurer, okay sure 😄 I will push changes in a moment.

File: .travis.yml (hunk 2 above)
Comment (tenshiAMD) [reply .travis.yml]: @manuelmeurer, sure 😄

File: lib/active_record/acts_as/querying.rb (hunk 1 above)
Comment (tenshiAMD) [reply lib/active_record/acts_as/querying.rb]: @manuelmeurer, I've made the changes, see commit https://github.com/krautcomputing/active_record-acts_as/pull/29/commits/645364b15e388442f764b5e35a54d9e1e61b625f. Thanks.

File: .travis.yml (hunk 2 above)
Comment (tenshiAMD) [reply .travis.yml]: @manuelmeurer, I've made the changes, see commit https://github.com/krautcomputing/active_record-acts_as/pull/29/commits/645364b15e388442f764b5e35a54d9e1e61b625f. Thanks.

Comment (tenshiAMD) [reply .travis.yml]: @manuelmeurer, it looks like ruby `2.3` also fails with rails `4.2` (https://travis-ci.org/krautcomputing/active_record-acts_as/jobs/475374092). In my opinion, we just need to support the latest stable versions of ruby (`2.4.5`, `2.5.3` and `2.6.0`) and since ruby `2.3.8` will be EOL soon acording to ruby site (https://www.ruby-lang.org/en/downloads/). What do you think?

Comment (tenshiAMD) [reply .travis.yml]: @manuelmeurer, If we still to support rails `4.2`, we need to limit `bundler` within `1.3` and below `2.0` as rails 4 bundler dependency(https://github.com/rails/rails/blob/v4.2.11/rails.gemspec#L30).

**Comment by manuelmeurer** [comment]
I dropped support for Ruby 2.3 and added Ruby 2.6 to the Travis config.
The only tests failing in master now are the ones for Rails 5.2, which your changes should fix.
Could you rebase against master once again?
Sorry for the back and forth.

**Comment by tenshiAMD** [comment]
no worries @manuelmeurer 😄 here https://github.com/krautcomputing/active_record-acts_as/commit/988e11f35c3311faf068ce66f8a21bf75a3e9e82 its done now 👍

**Review by manuelmeurer — APPROVED** [review]

**Comment by manuelmeurer** [comment]
Thanks, @tenshiAMD, fantastic work!! 😺
